### PR TITLE
processinfo: Add mmal support

### PR DESCRIPTION
**Pull request by popcornmix**
Just for comments.
I've hand edited the Makefile which needs handling properly.

I'm not sure if ProcessInfoPi is quite what you had in mind.

It doesn't actually work due to the filters/EINTERLACEMETHOD mixup.

### CODE REVIEW

File: xbmc/cores/VideoPlayer/Process/Makefile [hunk 1]
```diff
@@ -1,8 +1,9 @@
-INCLUDES+=-I/home/xbmc/progs/src/xbmc/xbmc/cores/VideoPlayer
+INCLUDES+=-I/home/dc4/xbmc_holder/xbmc/xbmc/cores/VideoPlayer
```
Comment (wsnipex) [inline xbmc/cores/VideoPlayer/Process/Makefile]: this cannot be hardcoded. Should be a Makefile.in

**Comment by popcornmix** [comment]
@wsnipex See first message https://github.com/xbmc/xbmc/pull/9154
I think the hope was that some advanced build stuff could create Makefile targets and set the define based on arbitrary files being present in the overrides directory.
As that's not present I'm just editing the Makefile directly (as @FernetMenta did) to test the code.

Any suggestions on how the implement the overrides detection in the build system would be welcome.

**Comment by wsnipex** [comment]
so the idea is to automatically include/build platform specific overrides whenever they are present in the overrides/$PLATFORM/ dir?
That can be done with e.g.
>SRCS := $(wildcard overrides/$(PLATFORM)/*.cpp)

for this to work as expected we also need to expose the PLATFORM names in kodi's configure, like we have in depends already.
CMake can do this as well.

**Comment by FernetMenta** [comment]
> so the idea is to automatically include/build platform specific overrides whenever they are present in the overrides/$PLATFORM/ dir?

correct.

**Pull request merged**